README.md

```diff
@@ -1,602 +1,387 @@
+
+If you _have_ modified the default config
+(or if you just want to be extra explicit),
+you can set the `:type` option at the top of the file, like so:
 
 ```ruby
-# rspec-rails-1.x
-assigns[key] = value
+# spec/models/user_spec.rb
 
-# rspec-rails-2.x+
-assign(key, value)
+RSpec.describe User, type: :model do
+...
 ```
 
-### `rendered`
+[modified the default `rails_helper.rb` configuration]: https://relishapp.com/rspec/rspec-rails/docs/upgrade#file-type-inference-disabled
 
-This represents the rendered view.
+### System specs, feature specs, request specs–what’s the difference?
 
-```ruby
-render
-expect(rendered).to match /Some text expected to appear on the page/
-```
-
-#### Upgrade note
+RSpec Rails provides three types of specs
+for integration testing the application as a whole—in other words,
+specifying what the client sees when interacting with it.
 
-```ruby
-# rspec-rails-1.x
-render
-response.should xxx
+#### System specs
 
-# rspec-rails-2.x+
-render
-rendered.should xxx
-
-# rspec-rails-2.x+ with expect syntax
-render
-expect(rendered).to xxx
-```
+Sometimes called **acceptance tests** or **browser tests**,
+system specs test the application from the perspective of a _human client._
+The test code walks through a user’s browser interactions,
 
-## Routing specs
+* `visit '/login'`
+* `fill_in 'Name', with: 'jdoe'`
 
-Routing specs default to residing in the `spec/routing` folder. Tagging any
-context with the metadata `:type => :routing` treats its examples as routing
-specs.
+and the expectations revolve around page content.
 
-```ruby
-require 'rails_helper'
-
-RSpec.describe "routing to profiles", :type => :routing do
-  it "routes /profile/:username to profile#show for username" do
-    expect(:get => "/profiles/jsmith").to route_to(
-      :controller => "profiles",
-      :action => "show",
-      :username => "jsmith"
-    )
-  end
+* `expect(page).to have_text('Welcome')`
 
-  it "does not expose a list of profiles" do
-    expect(:get => "/profiles").not_to be_routable
-  end
-end
-```
+Because system specs are a wrapper around Rails’ built-in `SystemTestCase`,
+they’re only available on Rails 5.1+.
 
-### Upgrade note
+#### Feature specs
 
-`route_for` from rspec-rails-1.x is gone. Use `route_to` and `be_routable`
-instead.
+Feature specs serve the same purpose,
+but `SystemTestCase` solves some longstanding configuration issues they had,
+so the RSpec team [officially recommends system specs][] over feature specs.
 
-## Helper specs
-
-Helper specs default to residing in the `spec/helpers` folder. Tagging any
-context with the metadata `:type => :helper` treats its examples as helper
-specs.
-
-Helper specs mix in ActionView::TestCase::Behavior. A `helper` object is
-provided which mixes in the helper module being spec'd, along with
-`ApplicationHelper` (if present).
+If you don’t have the luxury of upgrading,
+you’ll be limited to writing feature specs for your browser tests.
+They require [Capybara][],
```

Comment:
So do system specs, ... its not limiting, it just predates Rails setup for it.

Reply:
Fixed.